projects/novo-elements/src/elements/data-table/data-table-clear-button.component.ts

```diff
@@ -58,8 +60,13 @@ export class NovoDataTableClearButton<T> {
   }
 
   clearSearch(): void {
-    this.state.clearQuery();
-    this.queryClear.emit(true);
+    if (this.skipUpdate) {
+      this.queryClear.emit(true);
+      this.state.clearQuery(false);
```

Comment:
unless the ordering of these two calls matters, we can probably get rid of the if/else statement and just pass `!this.skipUpdate` into this function.